### ISSUE TITLE
Multiple splits are not properly output in order fetch and report

### ISSUE THREAD

**Issue by geradcoles** [geradcoles/touchbistro#1]
We currently join the ZORDER and ZPAIDORDER table, searching ZORDER for the ZORDERNUMBER column and using that to find a 1-limited paid order. That means only the first split is ever shown. The OrderFromId class needs to be broken up into orders and paid orders, where an order can have a one-to-many relationship to paid orders.

The behaviour in order fetch and report would be to output each paid order and include the ZI_SPLIT number and possibly any other useful split columns (like ZI_SPLITBY), to make it clear that we are looking at split 1 or 2 in an order split in half or what not. The CSV and JSON reports need to include the split index and related fields, as well. 

This solves the TODO I left in the OrderById class, where I raised concerns that multiple result rows was likely a problem.

**Comment by geradcoles** [comment]
Had to overhaul the Order object hierarchy considerably to accomplish this goal, but it was for the best as I was able to clean up some of the ugliest code in the order module by refactoring it. Order is now a list-type object containing all the splits associated with an order (when an order has no splits, it will be treated as a split with ID=0). Also fixed a bug where shared line-items would be missing from one of two orders that were split, when one of the two splits was closed, after which the other split continued to have additional items added. That was fixed through checking the ZPAIDORDER table for ZTABLEORDER and if found, getting order items for that ID, and concatenating the two sets of order items together.

**Comment by geradcoles** [comment]
I could not find a single historical order from January through the end of February 2020 where split order line item totals do not match PaidOrder object subtotals, demonstrating that line item math is fixed. I will continue to pour over the data as I did find a subtle void math issue related to order line items being split across seats, so there may still be obscure issues in the data. Closing for now.
